pkg/storage/storagewrappers/sharediterator/shared_iterator_datastore.go

```diff
@@ -113,21 +122,29 @@ func (s *storageItem) unwrap() (*sharedIterator, bool, error) {
 
 	s.once.Do(func() {
 		s.iter, s.err = s.producer()
+		if s.err != nil {
+			return
+		}
+
+		admissionTimer := time.AfterFunc(s.admissionDuration, func() {
+			s.iter.Stop()
```

Comment:
Why were `s.iter.Stop` calles moved into these timers if you originally closed it immediately (conditionally) after clone?